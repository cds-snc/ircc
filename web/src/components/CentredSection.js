import styled from 'react-emotion'
import PropTypes from 'prop-types'

const CentredSection = styled.section`
  margin-right: auto;
  margin-left: auto;
  width: ${props => (props.width ? props.width : '100%')};
`
CentredSection.propTypes = {
  width: PropTypes.string,
}

export default CentredSection
