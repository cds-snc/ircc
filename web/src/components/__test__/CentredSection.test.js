
import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import 'jest-styled-components'
import CentredSection from '../CentredSection'

describe('<CentredSection />', () => {
  it('can be rendered', () => {
    let wrapper = shallow(<CentredSection />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('accepts a width', () => {
    let wrapper = shallow(<CentredSection width="1000%"/>)
    expect(toJSON(wrapper)).toHaveStyleRule('width', '1000%')
  })
})
