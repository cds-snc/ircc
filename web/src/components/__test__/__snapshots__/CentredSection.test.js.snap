// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CentredSection /> accepts a width 1`] = `
<section
  className="css-p9npw3"
  width="1000%"
/>
`;

exports[`<CentredSection /> can be rendered 1`] = `
<section
  className="css-bo00q3"
/>
`;
